pyanaconda/payload/dnf/payload.py

```diff
@@ -684,15 +676,39 @@ def update_base_repo(self, fallback=True, checkmount=True):
                 return
 
             # Otherwise, fall back to the default repos that we disabled above
-            self._dnf_manager.restore_system_repositories()
-
-            # Enable or disable updates.
-            self.set_updates_enabled(self._updates_enabled)
+            self._enable_system_repositories()
 
         self._include_additional_repositories()
-        self._disable_unwanted_repositories()
         self._validate_enabled_repositories()
 
+    def _enable_system_repositories(self):
+        """Enable system repositories."""
+        self._dnf_manager.restore_system_repositories()
+
+        log.debug("Enable or disable updates repositories.")
+        self._set_repositories_enabled(conf.payload.updates_repositories, self._updates_enabled)
+
+        log.debug("Disable repositories based on the Anaconda configuration file.")
+        self._set_repositories_enabled(conf.payload.disabled_repositories, False)
```

Comment:
I don't think it's appropriate name of the method. It does not only enable system repositories but also disable them based on the Anaconda configuration.

I would change it more to `load_system_repositories` or something similar and mention this fact in the comment.

Reply:
Extended the doc string.